### PR TITLE
Baseline for error reporting window (won't do merge, just for review)

### PR DESCRIPTION
**Pull request by Deren-Liao**
Hi Ian, Ivan,

Most of the Error Reporting feature code is done. I split them into around 5 PRs, this is the first one.

I won't merge it into m2_master for now.  This is going to be the feature branch for error reporting window for m2 delivery.  I'll do changes on you comments and close the PR, leave the branch stay.  Later on, new PR's will target to this branch.

The PR contains 26 files --- The real content is not much!   Most of them are just empty place holders for now.  Content will be added in subsequent PR.  
(1) Data source code is in.
(2) Adds menu item, command class, window class , control class.  These are either VS generated files or just simply follow existing pattern. Control, view mode classes are now empty.  
(3) some icons that will be used later.

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+using Google.Apis.Clouderrorreporting.v1beta1.Data;
+using System.Collections.Generic;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// The result of Stackdriver error reporting list <seealso cref="ErrorEvent"/> request. 
+    /// </summary>
+    public class ErrorEventsRequestResult
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Do we need to create a new class for this? Can't we just use the current response type?  

Or is this what is happening in other places?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        /// <summary>
+        /// Gets a list of <seealso cref="ErrorEvent"/> objects. 
+        /// The value could be null or empty, caller to check.
+        /// </summary>
+        public IList<ErrorEvent> ErrorEvents { get; }
+
+        /// <summary>
+        /// A token is returned if available logs count exceeds the page size.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Can this be null too?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        public string NextPageToken { get; }
+
+        /// <summary>
+        /// Initializes an instance of <seealso cref="ErrorEventsRequestResult"/> class.
+        /// </summary>
+        /// <param name="errorEvents">
+        /// A list of <seealso cref="ErrorEvent"/> objects. 
+        /// Null is valid input.
+        /// </param>
+        /// <param name="pageToken">Next page token to retrieve more pages.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Can this be null?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+using Google.Apis.Clouderrorreporting.v1beta1.Data;
+using System.Collections.Generic;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// The result of Stackdriver error reporting list <seealso cref="ErrorGroupStats"/> request. 
+    /// </summary>
+    public class GroupStatsRequestResult
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs]: Same here can we just use the response object?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Clouderrorreporting.v1beta1.Data;
+using System.Collections.Generic;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// The result of Stackdriver error reporting list <seealso cref="ErrorGroupStats"/> request.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs]: Nit: Stackdriver Error Reporting

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Clouderrorreporting.v1beta1.Data;
+using System.Collections.Generic;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// The result of Stackdriver error reporting list <seealso cref="ErrorEvent"/> request.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Nit: Stackdriver Error Reporting

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+using Google.Apis.Clouderrorreporting.v1beta1.Data;
+using TimeRangeEnum = Google.Apis.Clouderrorreporting.v1beta1.ProjectsResource.GroupStatsResource.ListRequest.TimeRangePeriodEnum;
+using EventTimeRange = Google.Apis.Clouderrorreporting.v1beta1.ProjectsResource.EventsResource.ListRequest.TimeRangePeriodEnum;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// Data source that returns Google Cloud Stackdriver Error Reporting.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: returns what?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    /// </summary>
+    public class SerDataSource : DataSourceBase<ClouderrorreportingService>
+    {
+        /// <summary>
+        /// Initializes an instance of <seealso cref="SerDataSource"/> class.
+        /// </summary>
+        /// <param name="projectId">The project id that contains the GCE instances to manipulate.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: GCE? error reporting data?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        /// <param name="credential">The credentials to use for the call.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: param for appname?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        public SerDataSource(string projectId, GoogleCredential credential, string appName)
+                : base(projectId, credential, init => new ClouderrorreportingService(init), appName)
+        {}
+
+        /// <summary>
+        /// Get a list of <seealso cref="ErrorGroupStats"/> of the given <paramref name="groupId"/>.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: nit: gets

for the given

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        /// </summary>
+        /// <param name="timeRange">Specifiy the time range of the query.</param>
+        /// <param name="timedCountDuration">
+        /// The preferred duration for a single returned `TimedCount`. If not set, no timed counts are returned.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: "If not set"- This isn't an optional param.  Should this be? or should the comment change?

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        /// </param>
+        /// <param name="groupId">optional, The error group id.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Nit: Cap optional

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+        /// <param name="nextPageToken">optional, A next page token provided by a previous response.</param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Nit: Cap optional

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+            }
+            catch (GoogleApiException ex)
+            {
+                Debug.WriteLine($"Failed to get GroupStats: {ex.Message}");
+                throw new DataSourceException(ex.Message, ex);
+            }
+        }
+
+        /// <summary>
+        /// Request for a list <seealso cref="ErrorEvent"/> for an error group.
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: nit: requests

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+        /// </summary>
+        /// <param name="errorGroup">An error group. <seealso cref="ErrorGroupStats"/>.</param>
+        /// <param name="period">The time period for the query </param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Add optional and what it defaults too?

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.vsct [hunk 16]
```diff
@@ -106,9 +96,9 @@
         <Strings>
           <ButtonText>Publish project to Google Cloud...</ButtonText>
         </Strings>
-      </Button>
-
-        <!-- This command publishes the current project from the project's context menu. -->
+      </Button>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.vsct]: nit: added whitespace

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+    /// <summary>
+    /// Create a single instance of <seealso cref="SerDataSource"/>.
+    /// <seealso cref="ErrorReportingViewModel"/> and <seealso cref="ErrorReportingDetailViewModel"/> use the same instance.
+    /// </summary>
+    public static class SerDataSourceInstance
+    {
+        private static Lazy<SerDataSource> s_instance = new Lazy<SerDataSource>(CreateDataSource);
+
+        /// <summary>
+        /// Gets an instance of
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Finish comment?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerToolWindowCommand.cs [hunk 18]
```diff
@@ -30,7 +30,7 @@ internal sealed class LogsViewerToolWindowCommand
         /// <summary>
         /// Command ID.
         /// </summary>
-        public const int CommandId = 4231;
+        public const int CommandId = 4235;
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerToolWindowCommand.cs]: should this be changed here?

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+        public static ErrorReportingToolWindowCommand Instance
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Gets the service provider from the owner package.
+        /// </summary>
+        private IServiceProvider ServiceProvider
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs]: I know is generated code, but you could simplify it a bit, `public IServiceProvider ServiceProvider => package;`

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs [hunk 20]
```diff
@@ -0,0 +1,123 @@
...
+        {
+            get
+            {
+                return this.package;
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs]: Also rename the vars so they follow our convention.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+using GoogleCloudExtension.DataSources.ErrorReporting;
+using System;
+
+namespace GoogleCloudExtension.StackdriverErrorReporting
+{
+    /// <summary>
+    /// Create a single instance of <seealso cref="SerDataSource"/>.
+    /// <seealso cref="ErrorReportingViewModel"/> and <seealso cref="ErrorReportingDetailViewModel"/> use the same instance.
+    /// </summary>
+    public static class SerDataSourceInstance
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Again, use the expanded name.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+// limitations under the License.
+
+using GoogleCloudExtension.Accounts;
+using GoogleCloudExtension.DataSources.ErrorReporting;
+using System;
+
+namespace GoogleCloudExtension.StackdriverErrorReporting
+{
+    /// <summary>
+    /// Create a single instance of <seealso cref="SerDataSource"/>.
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: This is not hte ocmment for the class, seems more appropiate for a method. A comment for a class would start with `This class contains/represents ...`.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+        /// Gets an instance of 
+        /// </summary>
+        public static SerDataSource Instance => s_instance.Value;
+
+        /// <summary>
+        /// When the current project id is changed, the Instance needs to be recreated.
+        /// </summary>
+        public static void OnProjectIdChanged()
+        {
+            s_instance = new Lazy<SerDataSource>(CreateDataSource);
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: This is not necessary, you already declared `s_instance` to be statically initialized above.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs [hunk 24]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace GoogleCloudExtension.StackdriverErrorReporting
+{
+    /// <summary>
+    /// Create a single instance of <seealso cref="SerDataSource"/>.
+    /// <seealso cref="ErrorReportingViewModel"/> and <seealso cref="ErrorReportingDetailViewModel"/> use the same instance.
+    /// </summary>
+    public static class SerDataSourceInstance
+    {
+        private static Lazy<SerDataSource> s_instance = new Lazy<SerDataSource>(CreateDataSource);
```
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: This design seems very problematic to me, keep in mind that the if the user changes the current project or account you will have to recreate this data source. There's really no advantage to cache it. Re-create it every time the credential changes.

**Comment by ivannaranjo** [comment]
You shouldn't really base your PR on `m2_master`. You should branch from `m2_master` into a separate branch `m2_master_<something>` and then create a PR to merge your changes, from your private clone, into `m2_master_<something>`. Avoid using `m2_master` as a base unless you mean to merge into it. It is easy to get confused and merge to the wrong branch.

As we discussed you should be doing your development in your private branches in your personal clones of the main repo, not on the official repo anyway.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs (hunk 1 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: That makes sense too. I am changing it.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs (hunk 6 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/DataSourceBase.cs [hunk 25]
```diff
@@ -31,6 +31,11 @@ namespace GoogleCloudExtension.DataSources
     public abstract class DataSourceBase<TService> where TService : BaseClientService
     {
         /// <summary>
+        /// The projects/{ProjectId} string commonly used by Google Cloud APIs as project name.
+        /// </summary>
+        protected string ProjectIdQuery => $"projects/{ProjectId}";
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/DataSourceBase.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+using EventTimeRange = Google.Apis.Clouderrorreporting.v1beta1.ProjectsResource.EventsResource.ListRequest.TimeRangePeriodEnum;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace GoogleCloudExtension.DataSources.ErrorReporting
+{
+    /// <summary>
+    /// Data source that returns Google Cloud Stackdriver Error Reporting.
+    /// </summary>
+    public class SerDataSource : DataSourceBase<ClouderrorreportingService>
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 27]
```diff
@@ -0,0 +1,102 @@
...
+
+        /// <summary>
+        /// Request for a list <seealso cref="ErrorEvent"/> for an error group.
+        /// </summary>
+        /// <param name="errorGroup">An error group. <seealso cref="ErrorGroupStats"/>.</param>
+        /// <param name="period">The time period for the query </param>
+        /// <returns>
+        /// An async task with <seealso cref="ErrorEventsRequestResult"/> as result.
+        /// </returns>
+        public async Task<ErrorEventsRequestResult> ListEventsAsync(
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: I planed to add paging in follow up PR.  I think it makes sense to add now.  Added paging.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs (hunk 2 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: N/A

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs (hunk 3 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/ErrorEventsRequestResult.cs]: Removed. N/A.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs (hunk 5 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs]: Removed. N/A.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs (hunk 4 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/GroupStatsRequestResult.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 7 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: returns "Error Report(ing)" :)    --- I updated the comment.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 8 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Updated to The project id of Google Cloud

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 10 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 11 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Yes, optional.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 12 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 13 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 14 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Changed to Gets.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs (hunk 15 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+        /// <param name="timeRange">Specifiy the time range of the query.</param>
+        /// <param name="timedCountDuration">
+        /// The preferred duration for a single returned `TimedCount`. If not set, no timed counts are returned.
+        /// </param>
+        /// <param name="groupId">optional, The error group id.</param>
+        /// <param name="nextPageToken">optional, A next page token provided by a previous response.</param>
+        /// <returns>
+        /// Async task with <seealso cref="GroupStatsRequestResult"/> as result.
+        /// </returns>
+        public async Task<GroupStatsRequestResult> ListGroupStatusAsync(
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.DataSources/ErrorReporting/SerDataSource.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.vsct (hunk 16 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/GoogleCloudExtension.vsct]: Removed.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs (hunk 24 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Removed.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs (hunk 23 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Removed.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs (hunk 17 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Removed the class.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerToolWindowCommand.cs (hunk 18 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverLogsViewer/LogsViewerToolWindowCommand.cs]: This is intended change. This is to leave some range for context menu.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs (hunk 19 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs [hunk 29]
```diff
@@ -0,0 +1,123 @@
...
+                var menuCommandID = new CommandID(CommandSet, CommandId);
+                var menuItem = new OleMenuCommand(this.ShowToolWindow, menuCommandID);
+                commandService.AddCommand(menuItem);
+                Action setEnabled = () =>
+                {
+                    menuItem.Enabled = (CredentialsStore.Default?.CurrentAccount != null &&
+                        !string.IsNullOrWhiteSpace(CredentialsStore.Default?.CurrentProjectId));
+                };
+
+                setEnabled();
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs (hunk 20 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/ErrorReportingToolWindowCommand.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs (hunk 22 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: Removed the class.

File: GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs (hunk 21 above)
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension/StackdriverErrorReporting/SerDataSourceInstance.cs]: The class is removed.

**Comment by Deren-Liao** [comment]
Updated, please review.

**Comment by Deren-Liao** [comment]
Thanks for reviewing.

**Pull request closed without merging**